template-compose/data/src/test/java/co/nimblehq/template/compose/data/repository/AppPreferencesRepositoryImplTest.kt

```diff
@@ -0,0 +1,95 @@
+package co.nimblehq.template.compose.data.repository
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import androidx.datastore.preferences.preferencesDataStoreFile
+import androidx.test.core.app.ApplicationProvider
+import app.cash.turbine.test
+import co.nimblehq.template.compose.domain.repository.AppPreferencesRepository
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.io.IOException
+
+private const val TEST_DATASTORE_NAME: String = "test_datastore"
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+class AppPreferencesRepositoryImplTest {
+
+    private val testContext: Context = ApplicationProvider.getApplicationContext()
+    private val testCoroutineDispatcher = UnconfinedTestDispatcher()
+    private val testCoroutineScope = TestScope(testCoroutineDispatcher + Job())
+    private val testDataStore: DataStore<Preferences> = PreferenceDataStoreFactory.create(
+        scope = testCoroutineScope,
+        produceFile = { testContext.preferencesDataStoreFile(TEST_DATASTORE_NAME) }
+    )
+
+    private val mockDataStore: DataStore<Preferences> = mockk()
+    private lateinit var repository: AppPreferencesRepository
+
+    @Before
+    fun setup() {
+        repository = AppPreferencesRepositoryImpl(mockDataStore)
+    }
+
+    @Test
+    fun `When calling isFirstTimeLaunch request successfully, it returns correct response`() =
```

Comment:
we need to update the preference name, for example `When getting a preference` 💭